OMG.Classroom/API/Controllers/StudentController.cs

```diff
@@ -47,29 +47,23 @@ public async Task<IActionResult> DeleteAsync(Guid id)
             
         }
         [HttpPost]
-        public async Task<ActionResult<Student>> CreateAsync(Student student)
+        public async Task<IActionResult> CreateAsync(Student student)
         {
             //TODO - add mapping and change to ViewModel
             await _studentRepository.AddAsync(student);
             return CreatedAtAction(nameof(GetAsync), new { id = student.Id }, student);
         }
-        [HttpPut("id")]
-        public async Task<ActionResult<Student>> UpdateAsync(Guid id, Student student)
+        [HttpPut("{id}")]
+        public async Task<IActionResult> UpdateAsync(Guid id, Student student)
         {
             //TODO - add mapping and change to ViewModel
             if (student.Id != id)
```

Comment:
When updating, you don't pass the id in the body => you receive it as a parameter. Therefore, before updating you should set the id first and then pass student to the repo